### PR TITLE
feat: Implement infinite scrolling to load posts for Home page

### PR DESCRIPTION
**Pull request by leanhduy**
create separate Post component

remove some unused variables, libraries components on ImageUploadButton, Home

update the GraphQL client-side queries for fetching list of posts

update the server side's Nexus schema.js for adding query response type for feedForYou query

Closes P1-57

### CODE REVIEW

File: client/src/pages/home.js [hunk 1]
```diff
@@ -10,11 +10,15 @@ import { FEED_FOR_YOU, GET_USER_BY_ID } from '../utils'
 import { UserContext } from '../context'
 
 const Home = () => {
+    const [skip, setSkip] = useState(0)
```
Comment (pybotlad) [inline client/src/pages/home.js:13]: Provide some comment explaining the purpose of these step. Also organize the code by adding section using better comment convention
// * : For the information
// ? : For code explaining
// ! : For important notes
// TODO: For steps that can't be done now but need to be addressed when certain features is implemented.

File: client/src/pages/home.js [hunk 2]
```diff
@@ -10,11 +10,15 @@ import { FEED_FOR_YOU, GET_USER_BY_ID } from '../utils'
...
+    const [posts, setPosts] = useState([])
+    const ref = useRef()
+
     // TODO: OPTIMIZE TO REMOVE THE REDUNDANCY NEWPOSTDIALOG BETWEEN COMPONENTS
     const [isCreatingNewPost, setIsCreatingNewPost] = useState(false)
     // * Get the current logged in user
     // TODO: When authentication features are implement (login), remove this code and replace with the authenticated user (e.g., via Context API)
-    const { data: loggedInUser } = useQuery(GET_USER_BY_ID, {
+    useQuery(GET_USER_BY_ID, {
```
Comment (pybotlad) [inline client/src/pages/home.js:21]: Why there is no variables that will store the result of this useQuery. Then is this useQuery needed? If not, removed it.

File: client/src/pages/home.js [hunk 3]
```diff
@@ -29,51 +33,77 @@ const Home = () => {
     }
     // TODO: Replace this mockCurrentUser with logged in user via Context
     const mockCurrentUser = useContext(UserContext)
```
Comment (pybotlad) [inline client/src/pages/home.js:35]: Choose only one between fetching the user with id = 1 using `useQuery`, or using the hardcoded `mockCurrentUser`

File: client/src/pages/home.js [hunk 4]
```diff
@@ -29,51 +33,77 @@ const Home = () => {
...
-    const { loading, error, data, refetch } = useQuery(FEED_FOR_YOU)
+    const { data, refetch } = useQuery(FEED_FOR_YOU, {
+        variables: { skip: skip },
+    })
+
+    useEffect(() => {
```
Comment (pybotlad) [inline client/src/pages/home.js:40]: Add some explanation what the `useEffect` does, and for the code inside as well.

File: server/src/schema.js [hunk 5]
```diff
@@ -63,6 +63,14 @@ const AddPostResponse = objectType({
   },
 })
 
+const FeedQueryResponse = objectType({
```
Comment (pybotlad) [inline server/src/schema.js:66]: Add documentation comment to the type / query.

File: server/src/schema.js [hunk 6]
```diff
@@ -409,10 +417,15 @@ const Query = objectType({
     })
 
     // * Posts in "FOR YOU" mode
-    t.list.field('feedForYou', {
-      type: Post,
-      resolve: (_parent, _args, context) => {
-        return context.prisma.post.findMany({
+    t.field('feedForYou', {
```
Comment (pybotlad) [inline server/src/schema.js:420]: Add documentation comment

File: server/src/schema.js [hunk 7]
```diff
@@ -409,10 +417,15 @@ const Query = objectType({
...
+      type: 'FeedQueryResponse',
```
Comment (pybotlad) [inline server/src/schema.js:421]: Add documentation comment

**Review by pybotlad — CHANGES_REQUESTED** [review]
There are some changes need to be made:
1. There is duplication code for the simulated logged in user in home page (`useQuery` and `mockLogginUser`). Please update code to pick one and remove the other.
2. The query lacks documentation. Add comment to generate better documentation for the graphql queries and types.

Please address these issues and recommit the changes for next review. Thanks

File: client/src/pages/home.js (hunk 1 above)
Comment (leanhduy) [reply client/src/pages/home.js:13]: Comments provided.

File: client/src/pages/home.js (hunk 2 above)
Comment (leanhduy) [reply client/src/pages/home.js:21]: Remove the reference to the hardcoded user object in `mockData.js` and used the result of the GraphQL query `GET_USER_BY_ID`.
The logged user is still hardcoded (id = 1).

File: client/src/pages/home.js (hunk 4 above)
Comment (leanhduy) [reply client/src/pages/home.js:40]: Top-level explanation is provided for each `useEffect` hook

File: server/src/schema.js (hunk 6 above)
Comment (leanhduy) [reply server/src/schema.js:420]: Added code-level comment and graphql-level comment

File: server/src/schema.js (hunk 7 above)
Comment (leanhduy) [reply server/src/schema.js:421]: Duplicated. See the previous comment.

**Review by pybotlad — APPROVED** [review]
I'm happy with the changes. 
Go ahead and merge into `main` branch

**Pull request merged**